### PR TITLE
Order submodules and fix documentation headers

### LINKED ISSUE #179: Submodules ordering on the examples site
**Description**
The ITK examples index seems not to be ordered alphabetically at the submodule level
https://itk.org/ITKExamples/src/index.html

Although a search box exists, and the browsers search tool can be used to look for the module of interest, I'd dare to say that it would be nice if the submodules were ordered alphabetically.

### PR DESCRIPTION
**Pull request by mseng10**
## Description

1. This will index all of the submodules in alphabetical order when being displayed in Sphinx.
2. This will also index all of the examples in alphabetical order when being displayed in Sphinx. In order to make this easier for the developers to add new examples in alphabetical order, several examples had their folder names changed to the headers of their respective `Documentation.rst`.
3. Many of the examples do not have consistent casing in their `Documentation.rst` headers. This puts all of the examples in the standard title case format that satisfies MLA, APA, etc.
4. This includes the `titlecase` package in `CreateExample.py.in` to transform a folder name into an `rst` header. This will follow the proper casing mentioned earlier and will properly capitalize common abbreviations found in ITK.

Closes #179

### CODE REVIEW

File: src/Core/Common/CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,7 @@ add_example( CreateAnother )
 add_example( CreateAFixedArray )
 add_example( AddOffsetToIndex )
 add_example( GetNameOfClass )
-add_example( CreateAnImageRegion )
+add_example( CreateAImageRegion )
```
Comment (dzenanz) [inline src/Core/Common/CMakeLists.txt:23]: I think `an image` is correct. Is this a spelling change or to ease alphabetical sorting?

File: src/Filtering/Smoothing/index.rst [hunk 2]
```diff
@@ -4,10 +4,10 @@ Smoothing
...
-  ApplyMeanFilter/Documentation.rst
-  ApplyMedianFilter/Documentation.rst
   BlurringAnImageUsingABinomialKernel/Documentation.rst
   FindHigherDerivativesOfImage/Documentation.rst
+  MeanFilteringOfAnImage/Documentation.rst
+  MedianFilteringOfAnImage/Documentation.rst
   MedianFilteringOfAnRGBImage/Documentation.rst
   SmoothImageWithDiscreteGaussianFilter/Documentation.rst
-  SmoothWithRecursiveGaussian/Documentation.rst
+  ComputesSmoothingWithGaussianKernel/Documentation.rst
```
Comment (dzenanz) [inline src/Filtering/Smoothing/index.rst:13]: Doesn't this change make it non-alphabetical?

**Review by dzenanz — COMMENTED** [review]
Mostly looks good. Some remarks in-line.

**Review by thewtex — APPROVED** [review]
@mseng10 fantastic! :candy:

**Pull request merged**